src/core/component.js

```diff
@@ -332,7 +332,9 @@ function buildData (el, name, attrName, schema, elData, silent) {
     data = {};
     Object.keys(schema).forEach(function applyDefault (key) {
       var defaultValue = schema[key].default;
-      data[key] = typeof defaultValue === 'object' ? utils.extend({}, defaultValue) : defaultValue;
+      data[key] = defaultValue && defaultValue.constructor === Object
```

Comment:
I would be invert the conditions and be explicit with the check for null:
`defaultValue.constructor === Object && and defaultValue !== null`
In other parts of the code we use `typeof defaultValue === 'object' `. Is there any advantage on using `defaultValue.constructor === Object`. If they're equivalent I would use the first form for consistency.